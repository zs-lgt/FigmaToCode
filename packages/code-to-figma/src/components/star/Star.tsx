import * as React from 'react';
import {
    DefaultShapeProps,
    CornerProps,
    StarNodeProps,
    StyleOf,
    InstanceItemProps,
    SelectionEventProps
} from '../../types';
import {
    LayoutStyleProperties,
    transformLayoutStyleProperties
} from '../../styleTransformers/transformLayoutStyleProperties';
import {
    GeometryStyleProperties,
    transformGeometryStyleProperties
} from '../../styleTransformers/transformGeometryStyleProperties';
import { useYogaLayout } from '../../hooks/useYogaLayout';
import { transformBlendProperties, BlendStyleProperties } from '../../styleTransformers/transformBlendProperties';
import { YogaStyleProperties } from '../../yoga/YogaStyleProperties';
import { StyleSheet } from '../..';
import { useSelectionChange } from '../../hooks/useSelectionChange';
import { transformAutoLayoutToYoga } from '../../styleTransformers/transformAutoLayoutToYoga';
import { OnLayoutHandlerProps, useOnLayoutHandler } from '../../hooks/useOnLayoutHandler';
import { useImageHash } from '../../hooks/useImageHash';
import { useNodeIdCallback } from '../../hooks/useNodeIdCallback';

export interface StarProps
    extends DefaultShapeProps,
        CornerProps,
        StarNodeProps,
        InstanceItemProps,
        SelectionEventProps,
        OnLayoutHandlerProps {
    style?: StyleOf<YogaStyleProperties & LayoutStyleProperties & GeometryStyleProperties & BlendStyleProperties>;
    children?: undefined;
}

const Star: React.FC<StarProps> = props => {
    const nodeRef = React.useRef();

    useSelectionChange(nodeRef, props);
    useNodeIdCallback(nodeRef, props.onNodeId);

    const style = { ...StyleSheet.flatten(props.style), ...transformAutoLayoutToYoga(props) };

    const imageHash = useImageHash(style.backgroundImage);

    const starProps = {
        ...transformLayoutStyleProperties(style),
        ...transformGeometryStyleProperties('fills', style, imageHash),
        ...transformBlendProperties(style),
        ...props,
        style
    };
    const yogaProps = useYogaLayout({ nodeRef, ...starProps });
    useOnLayoutHandler(yogaProps, props);

    return <star {...starProps} {...yogaProps} innerRef={nodeRef} />;
};

export { Star };
