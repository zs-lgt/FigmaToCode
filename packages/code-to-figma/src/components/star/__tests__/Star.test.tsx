import * as React from 'react';
const renderer = require('react-test-renderer');
import { Star } from '../Star';

describe('<Star />', () => {
    it('Star without props', () => {
        const tree = renderer.create(<Star />).toJSON();
        expect(tree).toMatchSnapshot();
    });
});
