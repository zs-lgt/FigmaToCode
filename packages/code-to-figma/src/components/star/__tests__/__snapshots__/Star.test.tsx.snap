// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Star /> Star without props 1`] = `
<star
  innerRef={
    Object {
      "current": undefined,
    }
  }
  style={Object {}}
/>
`;
